packages/matchbox/src/components/Modal/Modal.tsx

```diff
@@ -33,26 +32,37 @@ const StyledFocusLock = styled(FocusLock)`
   ${focusLock}
 `;
 
-const StyledContent = styled(Box)`
+const StyledContent = styled(Box)<{ $state?: 'entered' | 'exiting' | 'exited' }>`
   ${content}
   ${contentAnimation}
 `;
 
-const Modal = React.forwardRef(function Modal(props, userRef) {
+export type ModalProps = {
+  children?: React.ReactNode;
+  className?: string;
+  closeOnEscape?: boolean;
+  closeOnOutsideClick?: boolean;
+  onClose?: () => void;
+  open: boolean;
+  portalId?: string;
```

Comment:
Small nitpick - but should this be pulled from `PortalProps`? the `containerId` type